staging/src/k8s.io/apiserver/pkg/storage/cacher/cacher.go

```diff
@@ -698,8 +698,22 @@ func computeListLimit(opts storage.ListOptions) int64 {
 	return opts.Predicate.Limit
 }
 
-func (c *Cacher) listItems(ctx context.Context, listRV uint64, key string, opts storage.ListOptions) (listResp, string, error) {
+func (c *Cacher) listItems(ctx context.Context, key string, opts storage.ListOptions) (listResp, string, error) {
 	if !opts.Recursive {
+		var listRV uint64
+		var err error
+		if opts.ResourceVersionMatch == "" && opts.ResourceVersion == "" {
```

Comment:
Handling consistent read non-recursive here because `WaitUntilFreshAndGet` is used by `GetList` and `Get`. Consistent read for get is simpler because we don't need to consider continue.

Reply:
Right, took the option 1 to move everything to bottom layer. I think it makes it safer with recursive as functions to handle List and GetList have same arguments, making it easy to make mistake.